Add error checks for 'constructor' username

jm90m opened the pull request:
Fixes #1723.
breaks - http://busy.org/@constructor 

Changes:
- was getting the `constructor` function in the user details objects rather than the actual username 'constructor' lol.. constructor is a reserved word

Sekhmet commented:
I would rather add prefix to the keys than to let users modify our objects.
There are many reserved keywords for Objects, and they can be overwritten by crafting account with specific user name, which can cause crashes or even vulnerabilities.

jm90m commented:
@Sekhmet good point, I added a helper in `stateHelpers` so we can use that to get the key when we store the user details in the state

Also updated activities / wallet page since they get affected by that username - https://busy.org/@constructor/transfers
https://busy.org/@constructor/activity
